File: lib/result/Ack.java
package com.teamlab.communication.wireless.lpr9201.result;

/**
 * リザルト Ack
 */
public class Ack extends Result {
	/**
	 * 
	 */
	public static final int RESULT_CODE = 0x81;

	/**
	 * 
	 */
	public static final int DATA_LENGTH_SIZE = 1; // byte

	/**
	 * 
	 * @param datas
	 */
	public Ack(int[] datas) {
		super(RESULT_CODE, DATA_LENGTH_SIZE, datas);
	}
}

File: lib/result/ack.js
/* jslint node: true */
"use strict"

/**
 * リザルト Ack
 */

exports.Ack = Ack;

/**
 * 結果コード
 */
Ack.RESULT_CODE = 0x81;

/**
 * データ長バイトサイズ
 */
Ack.DATA_LENGTH_BYTE_SIZE = 1; // byte

/**
 * @param {Result} result
 */
function Ack(result) {
  this.result = result;
}

/**
 * パース可能か
 * @param {Result} result
 */
Ack.canParse = function(result) {
  return result !== null && result.resultCode === Ack.RESULT_CODE;
}
